library/common/extensions/filters/http/test_read/filter.cc

```diff
@@ -0,0 +1,46 @@
+#include "library/common/extensions/filters/http/test_read/filter.h"
+
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace TestRead {
+
+Http::FilterHeadersStatus TestReadFilter::decodeHeaders(Http::RequestHeaderMap& request_headers,
+                                                        bool) {
+  // sample path is /failed?start=0x10000
+  std::basic_string_view<char> str = request_headers.Path()->value().getStringView();
```

Comment:
This operates solely on the plain string without any abstraction, does Envoy not have a more specific method to extract the parameters?

Also, I think you don't care at all about the name of the paramterer. Would it make sense to limit it to something like /failed?error=<error code>

Reply:
1. Thanks for pointing this out. Found envoy's utility lib for the query strings and used it.

2. I think it's best to leave as-is for now, changing it will introduce some unrelated changes. #1549 would probably need these parameter names. Will remove the need for it if we figure out #1549 doesn't need it.